src/metabase/api/user.clj

```diff
@@ -61,19 +61,15 @@
       (f user-or-id new-user-group-memberships)
       (maybe-set-user-permissions-groups! user-or-id (map :id new-user-group-memberships)))))
 
-(defn- updated-user-name [user-before-update first_name last_name]
-  (let [prev_first_name (:first_name user-before-update)
-        prev_last_name  (:last_name user-before-update)
-        first_name      (or first_name prev_first_name)
-        last_name       (or last_name prev_last_name)]
-    (when (or (not= first_name prev_first_name)
-              (not= last_name prev_last_name))
-      [(when-not (= first_name :explicitly-nil) first_name)
-       (when-not (= last_name :explicitly-nil) last_name)])))
-
-(defn- maybe-update-user-personal-collection-name! [user-before-update first_name last_name]
+(defn- updated-user-name [user-before-update changes]
+  (let [[previous current] (map #(select-keys % [:first_name :last_name]) [user-before-update changes])
+        new (merge previous current)]
+    (when (not= previous new)
+      (vals new))))
```

Comment:
The order of entries in (arbitrary) maps is not specified. It can happen that resulting sequence contains the last name before the first name. Depending on the input, the return value can be empty or contain one element only.

Reply:
Ah, good point. I've changed it so that the function returns the map. Then it can  be destructured with the keys, so the order is never relied upon.

Reply:
Also just changed the name from `new` to `updated-names` which avoids shadowing the `new` special form, like you said.